web-common/src/features/dashboards/url-state/convertExploreStateToURLSearchParams.ts

```diff
@@ -75,14 +80,19 @@ export function convertExploreStateToURLSearchParams(
   // But it is only available in TimeControlState and not MetricsExplorerEntity
   timeControlsState: TimeControlState | undefined,
   preset: V1ExplorePreset,
+  // Used to decide whether to compress or not based on the full url length
+  url: URL,
+  disableCompression = false,
```

Comment:
If this were provided as an object, it'd be more clear what it represents where it's called:
```
// init-embed-api.ts
convertExploreStateToURLSearchParams(
  dashboard,
  exploreSpec,
  timeControlsState,
  defaultExplorePreset,
  { disableCompression: true },
),
```

Reply:
Huh, that sounds bad. I'd expect developers to look at the function signature. Adding an object for a single optional params seems unnessary.

Reply:
Ok maybe it's overkill in this scenario, but I'm thinking of it similar to how TanStack's `queryOptions` are passed as an object, and all call sites are very readable.

Reply:
Hmmm probably makes sense in that case. There are a lot of params for in `queryOptions`, not having an object will make it ugly.